ci-automation/image.sh

```diff
@@ -115,33 +115,65 @@ function _image_build_impl() {
+    if [ "${MAJOR_A}" != "${MAJOR_B}" ]; then
+        case "${NEW_CHANNEL}" in
+          lts)
+            CHANNEL_A=stable
+            ;;
+          stable)
+            CHANNEL_A=beta
+            ;;
+          *)
+            CHANNEL_A=alpha
+            ;;
+        esac
+        VERSION_A=$(curl -fsSL --retry-delay 1 --retry 60 --retry-connrefused --retry-max-time 60 --connect-timeout 20 "https://${CHANNEL_A}.release.flatcar-linux.net/${BOARD_A}/current/version.txt" | grep -m 1 FLATCAR_VERSION= | cut -d = -f 2)
+    else
+        CHANNEL_A="${NEW_CHANNEL}"
+        VERSION_A="${NEW_CHANNEL_VERSION_A}"
+    fi
+    export VERSION_A
+    export CHANNEL_A
     export FROM_B="file://${PWD}/images/latest"
     # Use the directory directly (and BOARD_B and CHANNEL_B are unused)
     export VERSION_B="."
     echo "== Image differences compared to ${CHANNEL_A} ${VERSION_A} =="
-    rm -f package-diff
-    curl -fsSLO --retry-delay 1 --retry 60 --retry-connrefused --retry-max-time 60 --connect-timeout 20 "https://raw.githubusercontent.com/flatcar-linux/flatcar-build-scripts/master/package-diff"
-    chmod +x package-diff
+    NEW_VERSION=$(git tag --points-at HEAD)
+    cd ..
+    rm -rf flatcar-build-scripts
+    git clone "https://github.com/flatcar-linux/flatcar-build-scripts"
     echo "Package updates, compared to ${CHANNEL_A} ${VERSION_A}:"
-    FILE=flatcar_production_image_packages.txt ./package-diff "${VERSION_A}" "${VERSION_B}"
+    FILE=flatcar_production_image_packages.txt flatcar-build-scripts/package-diff "${VERSION_A}" "${VERSION_B}"
     echo
     echo "Image file changes, compared to ${CHANNEL_A} ${VERSION_A}:"
-    FILE=flatcar_production_image_contents.txt FILESONLY=1 CUTKERNEL=1 ./package-diff "${VERSION_A}" "${VERSION_B}"
+    FILE=flatcar_production_image_contents.txt FILESONLY=1 CUTKERNEL=1 flatcar-build-scripts/package-diff "${VERSION_A}" "${VERSION_B}"
     echo
     echo "Image kernel config changes, compared to ${CHANNEL_A} ${VERSION_A}:"
-    FILE=flatcar_production_image_kernel_config.txt ./package-diff "${VERSION_A}" "${VERSION_B}"
+    FILE=flatcar_production_image_kernel_config.txt flatcar-build-scripts/package-diff "${VERSION_A}" "${VERSION_B}"
     echo
     echo "Image file size change (includes /boot, /usr and the default rootfs partitions), compared to ${CHANNEL_A} ${VERSION_A}:"
-    FILE=flatcar_production_image_contents.txt CALCSIZE=1 ./package-diff "${VERSION_A}" "${VERSION_B}"
+    FILE=flatcar_production_image_contents.txt CALCSIZE=1 flatcar-build-scripts/package-diff "${VERSION_A}" "${VERSION_B}"
     echo
     BASE_URL="http://${BUILDCACHE_SERVER}/images/${arch}/${vernum}"
     echo "Image URL: ${BASE_URL}/flatcar_production_image.bin.bz2"
+    echo
+    # Provide a python3 command for the CVE DB parsing
+    export PATH="$PATH:$PWD/scripts/ci-automation/python-bin"
```

Comment:
Why not running `show-changes` inside the SDK container? It should have python and the packaging package already. Too much fuss with setting everything up?

Reply:
I thought about it but wasn't sure. Maybe we move this changelog printing to an own job and then it's faster to have it runnable without the SDK container.
The SDK container also adds all these
```
10:47:34  ###### Starting stopped container 'flatcar-images-amd64-9999.99.99-kai-print-changelog' ######
10:47:34  flatcar-images-amd64-9999.99.99-kai-print-changelog
10:47:34  usermod: no changes
10:47:34  bash: cannot set terminal process group (24): Inappropriate ioctl for device
10:47:34  bash: no job control in this shell
10:47:34  flatcar-images-amd64-9999.99.99-kai-print-changelog
10:47:34  
10:47:34  ###### Writing versionfile 'sdk_container/.repo/manifests/version.txt' to SDK '3292.0.0+nightly-20220706-2100', OS '9999.99.99+kai-print-changelog'. ######
10:47:34  
```
lines each time it is invoked^^

Reply:
Tried it, the SDK's python doesn't have pip nor the `packaging` module

Reply:
Interesting, because `packaging` is a dependency of portage, so it should be a part of SDK already (https://github.com/flatcar-linux/portage-stable/blob/main/dev-python/packaging/packaging-21.3-r2.ebuild).

Nevermind, we can tackle this later.